[Docs][LiveComponent] Add very simple download files docs (#2513)
symfony/ux: src/LiveComponent/doc/index.rst

    need to specify ``multiple`` attribute on HTML element and end ``name``
    with ``[]``.

.. _downloads:

Downloading files
-----------------

Currently, Live Components do not natively support returning file responses directly from a LiveAction. However, you can implement file downloads by redirecting to a route that handles the file response.

Create a LiveAction that generates the URL for the file download and returns a `RedirectResponse`::

        #[LiveAction]
        public function initiateDownload(UrlGeneratorInterface $urlGenerator): RedirectResponse
        {
            $url = $urlGenerator->generate('app_file_download');
            return new RedirectResponse($url);
        }

.. code-block:: html+twig

    <div {{ attributes }} data-turbo="false">
        <button
            data-action="live#action"
            data-live-action-param="initiateDownload"
        >
            Download
        </button>
    </div>

.. tip::

    When Turbo is enabled, if a LiveAction response redirects to another URL, Turbo will make a request to prefetch the content. Here, adding ``data-turbo="false"`` ensures that the download URL is called only once.

> .. note::
> 
>    Native support for file downloads in Live Components is under development. For more details, refer to the related `pull request #2483 <https://github.com/symfony/ux/pull/2483>`_.
> 

Reviewer: ```suggestion
```

Sorry i missed my preview the first time. I really think we should not tease in-development features, as we never know when we can release things, and how much time/energy will be required to do so :)

Author: I missed that too !

    need to specify ``multiple`` attribute on HTML element and end ``name``
    with ``[]``.

.. _downloads:

Downloading files
-----------------

Currently, Live Components do not natively support returning file responses directly from a LiveAction. However, you can implement file downloads by redirecting to a route that handles the file response.

Create a LiveAction that generates the URL for the file download and returns a `RedirectResponse`::

    use Symfony\Component\HttpFoundation\RedirectResponse;
    use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
    use Symfony\UX\LiveComponent\Attribute\LiveAction;

    // ...

    class MyDownloadButton
    {
    // ...

        #[LiveAction]
        public function initiateDownload(UrlGeneratorInterface $urlGenerator): RedirectResponse
        {
            $url = $urlGenerator->generate('app_file_download');
            return new RedirectResponse($url);

>         }

Reviewer: I'd say this is enough content here, just to illustrate. The documentation is already very long, let's focus on the valuable content (the paragraph before and the tip below)

>     use Symfony\Component\HttpFoundation\RedirectResponse;
>     use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
>     use Symfony\UX\LiveComponent\Attribute\LiveAction;
> 
>     // ...
> 
>     class MyDownloadButton
>     {
>     // ...
> 
>         #[LiveAction]
>         public function initiateDownload(UrlGeneratorInterface $urlGenerator): RedirectResponse
>         {
>             $url = $urlGenerator->generate('app_file_download');
>             return new RedirectResponse($url);
>         }
>     }

Reviewer 2: Some things were missing in the code example:

```suggestion
    // src/Twig/Components/FileDownload.php
    namespace App\Twig\Components;
    
    use Symfony\Component\HttpFoundation\RedirectResponse;
    use Symfony\UX\LiveComponent\Attribute\AsLiveComponent;
    use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
    use Symfony\UX\LiveComponent\Attribute\LiveAction;

    #[AsLiveComponent]
    class MyDownloadButton
    {
        use DefaultActionTrait;

        #[LiveAction]
        public function initiateDownload(UrlGeneratorInterface $urlGenerator): RedirectResponse
        {
            $url = $urlGenerator->generate('app_file_download');

            return new RedirectResponse($url);
        }
    }
```

.. code-block:: html+twig

    <div {{ attributes }} data-turbo="false">
        <button
            data-action="live#action"
            data-live-action-param="initiateDownload"
        >

>         Download

Reviewer 2: ```suggestion
            Download
```

        </button>
    </div>

> When Turbo is enabled, it will intercept the redirect and initiate a second request for the download URL. Adding `data-turbo="false"` ensures that the download URL is called only once.

Reviewer: ```suggestion

.. tip::

    When Turbo is enabled, if a LiveAction response redirects to another URL, Turbo will make a request to prefetch the content. Here, adding ``data-turbo="false"`` ensures that the download URL is called only once.
```

.. note::

>    Native support for file downloads in Live Components is under development. For more details, refer to the related `pull request #2483 <https://github.com/symfony/ux/pull/2483>`_.
> 

Reviewer: ```suggestion
   Native support for file downloads in Live Components is under development. For more details, refer to the related `pull request #2483 <https://github.com/symfony/ux/pull/2483>`_.

```

Let's not document what's not yet released, to avoid any frustration :)
